src/java.instrument/share/native/libinstrument/InvocationAdapter.c

```diff
@@ -486,17 +491,18 @@ jint loadAgent(JNIEnv* env, jstring path) {
         return JNI_ERR;
     }
 
-    // create JPLISAgent with JVMTI environment
-    if (createNewJPLISAgent(vm, &agent) != JPLIS_INIT_ERROR_NONE) {
-        return JNI_ERR;
-    }
-
     // get path to JAR file as UTF-8 string
     jarfile = (*env)->GetStringUTFChars(env, path, NULL);
     if (jarfile == NULL) {
         return JNI_ERR;
     }
 
+    // create JPLISAgent with JVMTI environment
+    if (createNewJPLISAgent(vm, &agent, jarfile, JNI_FALSE) != JPLIS_INIT_ERROR_NONE) {
```

Comment:
No warning is ever printed in this case since this is an agent being loaded from an executable JAR. Correct?

Reply:
That's right, it's just for agents that are loaded into a running VM.